src/lib/planviz/lqp_visualizer.cpp

```diff
@@ -54,25 +54,23 @@ void LQPVisualizer::_build_subtree(const std::shared_ptr<AbstractLQPNode>& node,
   }
 
   // Visualize subselects
-  if (const auto projection = std::dynamic_pointer_cast<ProjectionNode>(node)) {
-    for (const auto& column_expression : projection->column_expressions()) {
-      visit_expression(column_expression, [&](const auto& sub_expression) {
-        const auto lqp_select_expression = std::dynamic_pointer_cast<LQPSelectExpression>(sub_expression);
-        if (!lqp_select_expression) return ExpressionVisitation::VisitArguments;
+  for (const auto& expression : node->node_expressions()) {
```

Comment:
This does not seem to work when running `hyriseBenchmarkTPCH --visualize`, e.g., for Q20

Reply:
Ah, I see what you mean. will look into it.

Reply:
You cannot see the impl in the visualization, because of: `void TableScan::_on_cleanup() { _impl.reset(); }`

Now, we could of course re-create the impl upon visualization, but do we want that?

Reply:
It's in a string now.

Reply:
I think you resolved the wrong conversation. This one was originally about the visualization of subqueries being broken.

Reply:
Ah, sorry.

"Broken" as in "Impl unset" being displayed for TableScans in subqueries? I don't think we can easily fix that, it get's even harder for nested subqueries.

Reply:
I thought you were aware of subquery visualization lacking info such as runtimes, row counts etc. This is because the PQP that get's visualized is just the "blueprint" for the plans that get actually executed.

Reply:
The visualization does not show the subquery at all:

<img width="351" alt="screenshot 2018-10-19 at 22 46 30" src="https://user-images.githubusercontent.com/575106/47242986-dab97880-d3f0-11e8-9e91-8eebccb4ebbb.png">

Reply:
Now I got you. 

Subqueries were only shown when they were in Projections previously as well. Of course now this becomes an issue, since there is no Projection anymore... (subselect are visualized in the LQP btw, no matter which node they are in, thanks to AbstractLQPNode::node_expressions()) Lemme try to fix this.